### PR TITLE
Ezp 25820 add search button

### PR DESCRIPTION
**Pull request by StephaneDiot**
jira : https://jira.ez.no/browse/EZP-25820
## Description

Provides the search button in the discoveryBar and the adapted routing to a basic view

### CODE REVIEW

**Comment by StephaneDiot** [comment]
ping @dpobel

File: Resources/public/js/apps/plugins/ezsearch-routeappplugin.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+                name: "viewSearch",
+                path: "/ezsearch/search",
+                view: "ezserachSearchView",
+                sideViews: {'navigationHub': true, 'discoveryBar': false},
+                callbacks: ['open', 'checkUser', 'handleSideViews', 'handleMainView'],
+            });
+
+            app.on('*:viewSearchAction', function() {
+                app.navigateTo("viewSearch");
+                });
```
Comment (dpobel) [inline Resources/public/js/apps/plugins/ezsearch-routeappplugin.js]: indentation problem

File: Resources/public/js/apps/plugins/ezsearch-routeappplugin.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            var app = this.get('host');
+
+            app.views.ezsearchSearchView = {
+                type: Y.View,
+            };
+
+            app.route({
+                name: "viewSearch",
+                path: "/ezsearch/search",
+                view: "ezserachSearchView",
```
Comment (dpobel) [inline Resources/public/js/apps/plugins/ezsearch-routeappplugin.js]: typo here, and given the same typo is not repeated when declaring the view, this probably does not work very well :)

File: Resources/public/js/apps/plugins/ezsearch-routeappplugin.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    Y.eZSearch.Plugin.RouteAppPlugin = Y.Base.create('ezsearchRouteAppPlugin', Y.Plugin.Base, [], {
+        initializer: function () {
+            var app = this.get('host');
+
+            app.views.ezsearchSearchView = {
+                type: Y.View,
+            };
+
+            app.route({
+                name: "viewSearch",
```
Comment (dpobel) [inline Resources/public/js/apps/plugins/ezsearch-routeappplugin.js]: I would suggest to use `searchPrototype` as the name of this route

File: Resources/public/js/apps/plugins/ezsearch-routeappplugin.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+                path: "/ezsearch/search",
+                view: "ezserachSearchView",
+                sideViews: {'navigationHub': true, 'discoveryBar': false},
```
Comment (dpobel) [inline Resources/public/js/apps/plugins/ezsearch-routeappplugin.js]: I would keep the discovery bar visible as well

File: Resources/public/js/views/plugins/ezsearch-searchbuttonplugin.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
+YUI.add('ezsearch-searchbuttonplugin', function (Y) {
+    Y.namespace('eZSearch.Plugin');
+
+    Y.eZSearch.Plugin.SearchButtonPlugin = Y.Base.create('ezsearchSearchButtonPlugin', Y.Plugin.Base, [], {
+        initializer: function () {
+            var discoverybar = this.get('host'),
+                searchButton =  new Y.eZ.ButtonActionView({
+                    actionId: "viewSearch",
```
Comment (dpobel) [inline Resources/public/js/views/plugins/ezsearch-searchbuttonplugin.js]: same as https://github.com/ezsystems/platform-ui-search-prototype-bundle/pull/2/files#r65190595, I would use `searchPrototype` as id

**Comment by dpobel** [comment]
+1 when the various nitpicks are fixed

**Pull request merged**